libs/coin-evm/src/api/node/rpc.common.ts

```diff
@@ -132,7 +132,20 @@ export const getFeeData: NodeApi["getFeeData"] = currency =>
     const block = await api.getBlock("latest");
     const currencySupports1559 = Boolean(block.baseFeePerGas);
 
-    const feeData = await (async () => {
+    const feeData = await (async (): Promise<
+      | {
+          maxPriorityFeePerGas: BigNumber;
+          maxFeePerGas: BigNumber;
+          nextBaseFee: BigNumber;
+          gasPrice?: undefined;
```

Comment:
Yes we need to explicitly define each fiels because they are all accessed down bellow for the return of the `getFeeData` function.

https://github.com/LedgerHQ/ledger-live/blob/e24e469d6d417f3728b5962ca62ab0e28d75053c/libs/coin-evm/src/api/node/rpc.common.ts#L185-L192

If we update the return type of this inline async function:

https://github.com/LedgerHQ/ledger-live/blob/support/fix-ts-issues/libs/coin-evm/src/api/node/rpc.common.ts#L135-L148

To:
```ts
    const feeData = await (async (): Promise<
      | {
          maxPriorityFeePerGas: BigNumber;
          maxFeePerGas: BigNumber;
          nextBaseFee: BigNumber;
        }
      | {
          gasPrice: ethers.BigNumber;
        }
    > => {...}
```

We get a TS error when trying to access these properties because, for either type of the union, some properties are not defined.

For example we could get this error:
```
Property 'maxFeePerGas' does not exist on type '{ maxPriorityFeePerGas: BigNumber; maxFeePerGas: BigNumber; nextBaseFee: BigNumber; } | { gasPrice: BigNumber; }'.
  Property 'maxFeePerGas' does not exist on type '{ gasPrice: BigNumber; }'.
```

Here:

https://github.com/LedgerHQ/ledger-live/blob/e24e469d6d417f3728b5962ca62ab0e28d75053c/libs/coin-evm/src/api/node/rpc.common.ts#L186